server.go

```diff
@@ -330,12 +330,17 @@ func (s *Server) StartAsSubserver(lndClient lnrpc.LightningClient,
 		}
 	}()
 
-	// Start the macaroon service and let it create its default macaroon in
-	// case it doesn't exist yet.
-	if err := s.startMacaroonService(); err != nil {
-		return err
+	// If a macaroon string was passed in from lndclient, then we already 
+	// have a macaroon for Pool and don't need to start the macaroon
+	// service.
+	if macData == "" {
+		// Start the macaroon service and let it create its default 
+		// macaroon in case it doesn't exist yet.
+		if err := s.startMacaroonService(); err != nil {
```

Comment:
We can't just not start the macaroon service here. Otherwise we'll run into a nil pointer error in `ValidateMacaroon` if we're actually trying to validate an incoming one. Instead I think we need an additional parameter that specifies no macaroons should be created. Same goes for the other PRs in Loop and Faraday.

Reply:
Got it, my bad! Made that change when you get a chance to take a look. If this looks ok I'll make similar changes in the faraday and loop PRs as well.